docs/advanced/data_spec.md

```diff
@@ -9,8 +9,64 @@ DeepGNN supports both homogeneous and heterogeneous graphs. Nodes and Edges supp
 
 DeepGNN supports two file formats: JSON and TSV. Users can generate a graph in either format then our pipeline will convert it into binary for training and inference.
 
-1. [JSON](#json-format): Heterogeneous or homegeneous graph.
-2. [TSV](#tsv-format): Homogeneous graph only.
+1. [Linear](#linear-format): Heterogeneous or homegeneous graph.
+2. [JSON](#json-format): Heterogeneous or homegeneous graph.
+3. [TSV](#tsv-format): Homogeneous graph only.
+
+## Linear Format
+
+Here is the graph data Linear format. The format requires two files: graph.linear and meta.json.
+
+### Graph Data
+
+`graph.linear` layout
+
+```
+<node info> <edge_1_info> <edge_2_info> ...
```

Comment:
what avro/protopuf schemas can be used to decode graph in a linear format but with binary instead of text encoding?

Reply:
why do we want to introduce a new text only format? what benefits does it have in comparison to json/tsv if each line represents a node and all its outgoing edges.

Reply:
In another PR we can switch it so everything is on separate lines -- I've adde to this file "Linear is the most compact format, fastest to process by far and is most directly what is given to binary writers."

Reply:
Please, don't introduce a new format to break it in the next PR. The change is not incremental since single line node is not backward compatible with the node data split across multiple lines.